### PR TITLE
Top-k accuracy operator on host

### PR DESCRIPTION
**Pull request by slayton58**
Automatically copy from device -> host if necessary.

Thanks to @pooyadavoodi for the host top-k code.

### CODE REVIEW

**Comment by Yangqing** [comment]
Looks pretty cool, thanks Simon @slayton58 !

@bwasti Bram - maybe this could be a good pull request to test our sync scripts?

File: caffe2/operators/accuracy_op.cc [hunk 1]
```diff
@@ -14,21 +14,43 @@ bool AccuracyOp<float, CPUContext>::RunOnDevice() {
   DCHECK_EQ(label.dim32(0), N);
   Y->Resize(vector<TIndex>());
   const auto* Xdata = X.data<float>();
-  const auto* labeldata = label.data<int>();
+  const auto* labelData = label.data<int>();
+  const int top_k = top_k_;
```
Comment (akyrola) [inline caffe2/operators/accuracy_op.cc]: This code depends on top_k <= D. So I think we should add 

top_k = std::min(top_k, D)

Comment (akyrola) [inline caffe2/operators/accuracy_op.cc]: this local variable is not needed?

File: caffe2/python/cnn.py [hunk 2]
```diff
@@ -521,6 +521,22 @@ def Iter(self, blob_out, **kwargs):
             **kwargs)
         return self.net.Iter(blob_out, blob_out, **kwargs)
 
+    def Accuracy(self, blob_in, blob_out, **kwargs):
```
Comment (akyrola) [inline caffe2/python/cnn.py]: can we move this to model_helper.py instead, as this is not CNN specific function?

File: caffe2/operators/accuracy_op.cc (hunk 1 above)
Comment (Yangqing) [reply caffe2/operators/accuracy_op.cc]: Or better, do CAFFE_ENFORCE(top_k <= D, "Hey dude"); to guard against incorrect arguments.

Comment (pooyadavoodi) [reply caffe2/operators/accuracy_op.cc]: That's right, we can use `top_k_` instead.

Comment (pooyadavoodi) [reply caffe2/operators/accuracy_op.cc]: Done

Comment (pooyadavoodi) [reply caffe2/operators/accuracy_op.cc]: Done

File: caffe2/python/cnn.py (hunk 2 above)
Comment (pooyadavoodi) [reply caffe2/python/cnn.py]: I guess this should be done in different PR?

Comment (slayton58) [reply caffe2/python/cnn.py]: I can do this, but I won't be able to get to it for a little bit.

@pooyadavoodi it should be done in this PR, no sense in adding functionality and then immediately moving it.

Comment (slayton58) [reply caffe2/python/cnn.py]: @akyrola Moved to model_helper.py and rebased from master

**Pull request closed without merging**